src/nvim/getchar.c

```diff
@@ -2250,23 +2250,26 @@ static int get_key_from_user(int *timedoutp, int *mode_deletedp,
       setcursor();                /* put cursor back where it belongs */
   }
 
-  if (bytes_read < 0)
-    return -1;                     /* end of input script reached */
-  if (bytes_read == NUL) {                 /* no character available */
+  if (bytes_read < 0) {
+    /* end of input script reached will return -1 below */
+  } else if (bytes_read == NUL) {
```

Comment:
Thanks for the catch -- I'll fix that now. By way of explanation...

Checking for `NUL` is something left over from the original implementation that I should have changed when I split uses of a variable apart.
At this point a variable that represented a character was being reused to store the number of bytes read.
i.e. this test is checking that no bytes have been read.